modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/objectbuilder/DefaultObjectBuilder.java

```diff
@@ -121,28 +116,12 @@ public boolean isDynamic() {
   public T build(ValueResolvingContext context) throws MuleException {
     T object = createInstance(prototypeClass);
 
-    final CursorComponentDecoratorFactory componentDecoratorFactory =
-        context != null
-            ? (CursorComponentDecoratorFactory) context.getProperty(COMPONENT_DECORATOR_FACTORY_KEY)
-            : null;
-
     for (Map.Entry<FieldSetter, ValueResolver<Object>> entry : resolvers.entrySet()) {
       final Object resolvedValue = resolveValue(entry.getValue(), context);
 
       entry.getKey().set(object,
-                         context == null || context.resolveCursors()
-                             ? resolveCursor(resolvedValue,
-                                             entry.getValue().isContent() && componentDecoratorFactory != null
-                                                 ? v -> decorateInput(v, context.getEvent()
-                                                     .getCorrelationId(),
-                                                                      componentDecoratorFactory)
-                                                 : identity())
-                             : entry.getValue().isContent() && componentDecoratorFactory != null
-                                 ? mapTypeValue(resolvedValue,
-                                                v -> decorateInput(v, context.getEvent()
-                                                    .getCorrelationId(),
-                                                                   componentDecoratorFactory))
-                                 : resolvedValue);
+                         context == null || context.resolveCursors() ? resolveCursor(resolvedValue, MessageUtils::decorateInput)
+                             : resolvedValue);
```

Comment:
Note that now we are mapping with `MessageUtils#decorateInput` every time we do cursor resolution, regardless of the value being content or not. This is because `CursorStream`s need to be wrapped in `UnclosableCursorStreams` (see #11209). Please check if this is correct, because maybe I'm doing it in cases in which we shouldn't.

Reply:
pretty sure this case shuold not be wrapped. @dabreu @ndinu  can you confirm?

Reply:
I think that with the processing done in MethodArgumentResolverDelegate, the case is covered.

Reply:
That's great! I can also confirm that if I only do the wrapping on the `MethodArgumentResolverDelegate`, the test cases pass, but wasn't sure if it was a matter of lack of coverage.

Reply:
@ndinu can you point to/add that fills the gap between `i think` and `i'm sure`? :)

Reply:
This issue has now been addressed in master by #11333. We added a few missing test cases and made sure the `UnclosableCursorStream` wrapping was done regardless of the parameter being annotated as `@Content` or not.

I rebased this PR on top of master and adjusted accordingly.